### PR TITLE
APPS-162 Delete emails

### PR DESCRIPTION
**Pull request by jonasg-gc**

### CODE REVIEW

File: scripts/APPS-162-delete-emails.js [hunk 1]
```diff
@@ -0,0 +1,315 @@
...
+	
+}
+
+async function pull(coll, topicIds, arrayToPullFrom) {
+	await db.collection(coll).update(
+		{},
+		{ 
+			$pull:
+				{
+					arrayToPullFrom:
```
Comment (duboisp) [inline scripts/APPS-162-delete-emails.js:305]: This don't set the variable, it only define the key to be `arrayToPullFrom`

And after I tried to fix that in my local test, it didn't updated the entry. I found out the option `multi: true` was needed.

So I suggest the following code for this function

```js
let updateQuery = { $pull: { } };
updateQuery[ "$pull"][ arrayToPullFrom ] = { topicId: { $in: topicIds } };

await db.collection( col1 ).update(
  {},
  updateQuery,
  { multi: true }
);
```

After that change, it was working as expected and I got the "DONE!" success message.

**Review by duboisp — CHANGES_REQUESTED** [review]
I reviewed the code and I found a few parsing error and a run time error.

See my suggestion inline which after your script was working as expected.

Let me know when those change are completed, I will run another local test.

Thanks and you did a great work

File: scripts/APPS-162-delete-emails.js [hunk 2]
```diff
@@ -0,0 +1,315 @@
...
+	}
+	
+	let subsExistEmailCount = await db.collection("subsExist").countDocuments({t: {$in: topicIds}});
+	console.log(`found ${subsExistEmailCount} entries for the specified topicIds in subsExist`);
+	await db.collection("subsExist").deleteMany({t: {$in: topicIds}}); 
+	console.log(`Deleted entries for the specified topicIds in subsExist`);
+	subsExistEmailCount = await db.collection("subsExist").countDocuments({t: {$in: topicIds}});
+	console.log(`found ${subsExistEmailCount} entries for the specified topicIds in subsExist`);
+	
+	console.log(chalk.green("✓ DONE!"));
```
Comment (duboisp) [inline scripts/APPS-162-delete-emails.js:64]: Missing the removal of the topic from the collection "topic" and "topics_details". Or at least to invalidate its notifyKey + confirmURL + unsubURL + failURL in the "topic" collection to ensure that topic is deactivated and out of order.

**Review by duboisp — CHANGES_REQUESTED** [review]

File: scripts/APPS-162-delete-emails.js [hunk 3]
```diff
@@ -0,0 +1,315 @@
...
+								}
+						   }
+					  }
+					  
+				 ]
+		}
+	);
+	
+	if(numSubLogsWithSpecTopicIds > 0) {
+		throw new Error("We didn't delete all the sub logs for the specified topic IDs");
```
[the remark this answers is not in the document]
Comment (jonasg-gc) [reply scripts/APPS-162-delete-emails.js:293]: I went thru all error messages and console logs and made them much more helpful. :)

File: scripts/APPS-162-delete-emails.js (hunk 1 above)
Comment (jonasg-gc) [reply scripts/APPS-162-delete-emails.js:305]: I updated my pull function to match your logic.  Thanks @duboisp !

Comment (jonasg-gc) [reply scripts/APPS-162-delete-emails.js:305]: Note: This is one of the complications of moving from Mongo Shell queries with values copy-and-pasted to a node script with dynamic values. 😛

File: scripts/APPS-162-delete-emails.js (hunk 2 above)
Comment (jonasg-gc) [reply scripts/APPS-162-delete-emails.js:64]: Hmmm...  The request was just to delete the e-mail addresses, not remove any trace of the topic, entirely.  Perhaps they might wanna keep _some_ record of the topic having existed?

Comment (jonasg-gc) [reply scripts/APPS-162-delete-emails.js:64]: I went with your second suggestion for now.  See the _invalidateTopics_ function.

**Comment by jonasg-gc** [comment]
@duboisp : Thanks for the review.  I've addressed all your comments so far.  Please re-review.  Thanks! :)

**Review by duboisp — APPROVED** [review]
Review and tested locally,

Seems to fully work as expected.

Great work on this @jonasg-gc

**Pull request merged**